### PR TITLE
Write program for webserver to act like a faucet service

### LINKED ISSUE #1048: Write program for webserver to act like a faucet service
The program should keep running and server a POST request for anyone to request BAND. The program utilize the library we have for sending transactions.

### PR DESCRIPTION
**Pull request by Benzbeeb**
Fixed #1048

### CODE REVIEW

File: chain/cmd/requestband/main.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	if tmp == "" {
+		return def
+	}
+	return tmp
+}
+
+var (
+	port        = getEnv("PORT", "5001")
+	nodeURI     = getEnv("NODE_URI", "http://localhost:26657")
+	queryURI    = getEnv("QUERY_URI", "http://localhost:1317")
```
Comment (taobun) [inline chain/cmd/requestband/main.go]: Remove this variable

File: chain/cmd/requestband/main.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	priv        = getEnv("PRIVATE_KEY", "27313aa3fd8286b54d5dbe16a4fbbc55c7908e844e37a737997fc2ba74403812")
+	sandboxMode = os.Getenv("SANDBOX_MODE") != ""
```
Comment (taobun) [inline chain/cmd/requestband/main.go]: ditto

File: chain/cmd/requestband/main.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+)
+
+var rpcClient *rpc.HTTP
```
Comment (taobun) [inline chain/cmd/requestband/main.go]: ditto

File: chain/cmd/requestband/main.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	rpc "github.com/tendermint/tendermint/rpc/client"
+)
+
+type Request struct {
+	Address string `json:"address" binding:"required"`
+	Amount  int64  `json:"amount" binding:"required"`
+}
+type Response struct {
+	Result string `json:"result"`
```
Comment (taobun) [inline chain/cmd/requestband/main.go]: Change to TxHash

File: chain/cmd/requestband/main.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+		ToAddress:   to,
+		Amount:      sdk.NewCoins(sdk.NewCoin("uband", sdk.NewInt(req.Amount))),
+	}, 1000000, "", "")
+
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+	c.JSON(200, Response{
+		Result: result.String(),
```
Comment (taobun) [inline chain/cmd/requestband/main.go]: TxHash

File: chain/cmd/requestband/main.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+	})
+}
+
+func main() {
+	privBytes, _ := hex.DecodeString(priv)
+	copy(pk[:], privBytes)
+
+	var err error
+	bandClient, err = bandlib.NewBandStatefulClient(nodeURI, pk, 100, 10, "request band requests")
```
Comment (taobun) [inline chain/cmd/requestband/main.go]: Band faucet

File: chain/cmd/requestband/main.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+	if err != nil {
+		panic(err)
+	}
+	rpcClient = rpc.NewHTTP(nodeURI, "/websocket")
```
Comment (taobun) [inline chain/cmd/requestband/main.go]: Remove it

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — CHANGES_REQUESTED** [review]
Let's add it to docker-compose and varnish

**Review by taobun — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**